### PR TITLE
Run3-gex108XB Backport SimCore/SimG4CMS related files which can work with the new Run3 geometry

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Backport SimCore/SimG4CMS related files which can work with the new Run3 geometry

#### PR validation:

Use the runTheMatrix test workflows

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Backporting the PR's #36652, #36680, #36750, #36411, #36524 which were used to introduce the new Run3 geometry

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
Backport of #36750, #36652
(please check, and update the PR description with it)

**Comment by bsunanda** [comment]
@perrotta I took care of all the comments. You may seee some corrections because I started with a build which was before merging #36524

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
@bsunanda this PR backports also #36680 (Correction to some of the Forward detector SD's for the dd4hep scenarios). I imagine this is wanted: if so, could you please add it to the list of the backported PRs in the overall description, together with an explanation about why it is needed.

Moreover: would you mind rebasing on top of some recent 12_2_X IB, in order to ease the comparison with the original PRs intended to be backported? (We are backporting a large amoint of code into a suppposedly stable production release, better to have the possibility to double check that there are no mistakes)

**Comment by bsunanda** [comment]
Backporting PR #36680 was necessary and I have added that in the list of {R's to be back ported

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
I rebased this with the latest IB of CMSSW_12_2_X

**Comment by civanch** [comment]
@bsunanda , I would agree with @perrotta  that in 12_2 better to have PPS disabled. Concerning Exception backport: all classes in SimG4Core/Applications and SimG4Core/Notification should be taken. If we take some part of 12_3 modifications related to Exception, then we may be in troubles. More safe would be do not backport things related to Exception.

**Comment by bsunanda** [comment]
@civanch I shall make a separate backport request concerning exceptions. Could you please let me know which PR's were used for that. PPS is disabled for Run3 in this current version

**Comment by civanch** [comment]
@bsunanda , there are two PRs: #36717 and #36737 but I would propose do not do this backport, because 12_2 production may go with old exception scheme. It is not relevant to 12_2 production campaigns.

**Comment by civanch** [comment]
@bsunanda , please, resolve Andrea comments and let us merge this PR. We should not do more - the same exception scheme was in Run-2, it is working. Improvement for 12_3 is useful but not for the backport.

**Comment by bsunanda** [comment]
@civanch I took care of Andrea's comments in rebasing this PR and making relevant changes in the cfi's. There may be changes to some comment lines which I didn't bother to change - the comments are relevant to the current version of the code

File: SimG4Core/Application/python/g4SimHits_cfi.py [hunk 1]
```diff
@@ -652,9 +665,12 @@
...
+                       OnlySDs = ['ZdcSensitiveDetector', 'TotemT2ScintSensitiveDetector', 'TotemSensitiveDetector', 'RomanPotSensitiveDetector', 'PLTSensitiveDetector', 'MuonSensitiveDetector', 'MtdSensitiveDetector', 'BCM1FSensitiveDetector', 'EcalSensitiveDetector', 'CTPPSSensitiveDetector', 'HGCalSensitiveDetector', 'BSCSensitiveDetector', 'CTPPSDiamondSensitiveDetector', 'FP420SensitiveDetector', 'BHMSensitiveDetector', 'HFNoseSensitiveDetector', 'HGCScintillatorSensitiveDetector', 'CastorSensitiveDetector', 'CaloTrkProcessing', 'HcalSensitiveDetector', 'TkAccumulatingSensitiveDetector'],
+                       MuonSD = dict( 
+                       HaveDemoChambers = False )
```
Comment (perrotta) [inline SimG4Core/Application/python/g4SimHits_cfi.py:675]:
```suggestion
                       OnlySDs = ['ZdcSensitiveDetector', 'TotemT2ScintSensitiveDetector', 'TotemSensitiveDetector', 'RomanPotSensitiveDetector', 'PLTSensitiveDetector', 'MuonSensitiveDetector', 'MtdSensitiveDetector', 'BCM1FSensitiveDetector', 'EcalSensitiveDetector', 'CTPPSSensitiveDetector', 'HGCalSensitiveDetector', 'BSCSensitiveDetector', 'CTPPSDiamondSensitiveDetector', 'FP420SensitiveDetector', 'BHMSensitiveDetector', 'HFNoseSensitiveDetector', 'HGCScintillatorSensitiveDetector', 'CastorSensitiveDetector', 'CaloTrkProcessing', 'HcalSensitiveDetector', 'TkAccumulatingSensitiveDetector'],
                       LHCTransport = False, 
                       MuonSD = dict( 
                       HaveDemoChambers = False ) 
```

**Review by perrotta — COMMENTED** [review]
This is the only remaining difference I found with respect to the cumulative backport of 
- #36750 Simhits for demo chambers
- #36652: Make runtime variables to allow/disallow creation of SimHits for demonstration chambers
- #36680: Correction to some of the Forward detector SD's for the dd4hep scenarios
- #36411: Enable PPS simulation for Run-3
- #36524: Safer syntax for modifying the configuration parameters in g4SimHits_cfi.py

@bsunanda please apply

**Comment by perrotta** [comment]
@bsunanda please fix the PR description field with the exact PR numbers which are backported (see previous comment)

**Comment by perrotta** [comment]
@bsunanda @cvuosalo please agree about the opportunity of backporting also #36680, as it is done now.
Once https://github.com/cms-sw/cmssw/pull/36780#pullrequestreview-871504903 will get cured, that is the only remaining issue with this backport PR.

**Comment by bsunanda** [comment]
Without PR #36680 SD's of the forward detectors will not work for dd4hep. The forward detectors are now part of Run3 scenario and dd4hep is the default geometry base for Run3. So this PR #36680 must be back ported to12_2_X

**Comment by perrotta** [comment]
> Without PR #36680 SD's of the forward detectors will not work for dd4hep. The forward detectors are now part of Run3 scenario and dd4hep is the default geometry base for Run3. So this PR #36680 must be back ported to12_2_X

Thank you Sunanda.
If @civanch also agree for me it's fine
Please cure https://github.com/cms-sw/cmssw/pull/36780#discussion_r798322453 anyhow

**Comment by perrotta** [comment]
> @perrotta Which fix to be cured - do you mean remove the line LHCTransport = False for phase2 modification?

No: to add it, as now it is missing

**Comment by bsunanda** [comment]
@perrotta Only one thing I did in this update: added back the "LHCTransport = False" in g4SimHits_cfi.py

**Comment by perrotta** [comment]
> @perrotta Only one thing I did in this update: added back the "LHCTransport = False" in g4SimHits_cfi.py

Thank you @bsunanda 
Now the backport fully reflects what is merged in the master

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**